saleor/attribute/models/base.py

```diff
@@ -242,23 +242,27 @@ def bulk_get_or_create(self, objects_data):
         results = []
         objects_not_in_db: List[AttributeValue] = []
 
+        # prepare a list that will save order index of attribute values
+        objects_enumerated = [
+            (index, obj_data) for index, obj_data in enumerate(objects_data)
+        ]
         query = self._prepare_query_for_bulk_operation(objects_data)
 
         # iterate over all records in db and check if they match any of objects data
         for record in query.iterator():
             # iterate over all objects data and check if they match any of records in db
-            for obj in objects_data:
+            for index, obj in objects_enumerated:
                 if self._is_correct_record(record, obj):
                     # upon finding existing record add it to results
-                    results.append(record)
-
-                    # remove it from objects data, so it won't be added to new records
-                    objects_data.remove(obj)
-
-                    break
```

Comment:
This line should be reverted. There won't be more matching record, so there is no point in iterating over rests of the records.